dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,197 @@ public static void prepare() throws Exception {
+        String uri = "/test1_" + i;
+        String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
+        int action = 200;
+        int order = 1;
+
+        contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+
+        
+        
+        Optional<CachedVanityUrl> optionalCache = cache.getDirectMapping(uri, defaultHost, defaultLanguage);
+        assert(optionalCache==null );
+        
+
+        Optional<CachedVanityUrl> vanityCached = vanityUrlAPI.resolveVanityUrl(uri, defaultHost, defaultLanguage);
+
+        // it is not here because it is not published
+        assert(!vanityCached.isPresent());
+
+            
+        // it is stored in the cache as an Optional.empty();
+        optionalCache = cache.getDirectMapping(uri, defaultHost, defaultLanguage);
+        
+        assert(optionalCache!=null && !optionalCache.isPresent());
+            
     }
 
+
     /**
-     * Testing {@link VanityUrlAPI#getVanityUrlFromContentlet(com.dotmarketing.portlets.contentlet.model.Contentlet)}
+     * Creates a couple of vanity urls, publish only the first one.
+     * Checks that the first one is in cache and the second one is not.
      *
      * @see VanityUrlAPI
      */
     @Test
-    public void getVanityUrlFromContentlet() {
-        Contentlet contentlet1 = null;
-        try {
-            long i = System.currentTimeMillis();
-            String title = "VanityURL" + i;
-            String site = defaultHost.getIdentifier();
-            String uri = "/test3_" + i;
-            String forwardTo = "/about-us/" + CMSFilter.CMS_INDEX_PAGE;
-            int action = 301;
-            int order = 1;
-
-            contentlet1 = filtersUtil.createVanityUrl(title, site, uri,
-                    forwardTo, action, order, defaultLanguageId);
-
-            filtersUtil.publishVanityUrl(contentlet1);
-
-            VanityUrl vanity1 = vanityUrlAPI
-                    .getVanityUrlFromContentlet(contentlet1);
-
-            Assert.assertNotNull(vanity1);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL,vanity1.getInode());
-            Assert.assertEquals(title, vanity1.getTitle());
-            Assert.assertEquals(site, vanity1.getSite());
-            Assert.assertEquals(uri, vanity1.getURI());
-            Assert.assertEquals(forwardTo, vanity1.getForwardTo());
-            Assert.assertEquals(action, vanity1.getAction());
-
-        } catch (DotDataException | DotSecurityException e) {
-            e.printStackTrace();
-            fail();
-        } finally {
-            try {
-                if (contentlet1 != null) {
-                    contentletAPI.destroy(contentlet1, user, false );
-                }
-            } catch (Exception e) {
-                Logger.error(this.getClass(), "Error cleaning up Vanity Url Links");
-            }
-        }
+    public void vanities_resolve_once_they_are_published() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)